### PR TITLE
Major QoL update

### PR DESCRIPTION
**Pull request by friedkiwi**
This change adds:

* documentation
* functioning MQTT support
* basic web-frontend

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Gatekeeper
+
+This application is responsible for opening and closing the front gate.
+
+## Context
+
+Gatekeeper is a raspberry pi (hostname `zuul`) that is responsible for opening the front gate for guest and members. The pi is used to control the gate remote control and uses a Sim800 module to receive calls
```
Comment (evils) [inline README.md:7]:
```suggestion
Gatekeeper is a raspberry pi (hostname `zuul`) that is responsible for opening the front gate for guests and members. The pi is used to control the gate remote control and uses a Sim800 module to receive calls
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Usage
+
+The gatekeeper application uses a whitelist in the following format:
+
+    +123456 owner
+
+The first element is the phone number in international notation without spaces or other punctuation, the second element is a free text field describing the owner of the phone number.
```
Comment (evils) [inline README.md:15]: the first 4 characters of the free text field may be used on the flappers (though that currently doesn't work)

File: gatekeeper.service [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Service to open roll-up door upon phone calls
+
+[Service]
+User=pi
+Group=pi
+ExecStart=/usr/bin/python3 /home/pi/gatekeeper/main.py -d /home/pi/gatekeeper/whitelist -m mqtt --journald -v
+
+[Install]
+WantedBy=multi-user.target
```
Comment (evils) [inline gatekeeper.service:10]: missing newline at end of file

File: static/opened.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    <body>
+        <div class='centerbox'>
+            <img src="HSG3.png" alt="hsg logo" class='logo' />
+            <h1 class="centertext">WebGatekeeper</h1>
+            <p class="centertext bold large">
+                Gate opened
+            </p>
+        </div>
+    </body>
+</html>
```
Comment (evils) [inline static/opened.html:17]: missing newline at end of file

File: static/index.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    <body>
+        <div class='centerbox'>
+            <img src="HSG3.png" alt="hsg logo" class='logo' />
+            <h1 class="centertext">WebGatekeeper</h1>
+            <p class="centertext bold large">
+                <a href="/open_sesame">Open exterior gate</a>
+            </p>
+        </div>
+    </body>
+</html>
```
Comment (evils) [inline static/index.html:17]: missing newline at end of file

File: static/style.css [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    text-align: center;
+}
+
+.bold {
+    font-weight: bold;
+}
+
+.large {
+    font-size: x-large;
+}
```
Comment (evils) [inline static/style.css:34]: missing newline at end of file

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1 @@
+whitelist
```
Comment (evils) [inline .gitignore:1]: missing newline at end of file

File: README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+Gatekeeper can also be monitored and controlled using MQTT.
+
+Gatekeeper listens to commands on the `hsg/gatekeeper/cmd` topic. So far it supports the following commands:
+
+* `open`: open the gate
+
+Gatekeeper will also publish events on the `hsg/gatekeeper/*` topics:
+
+* `hsg/gatekeeper/open`: this topic will have a message published with the owner as described in the data file upon a successful call to open the gate.
```
Comment (evils) [inline README.md:33]: it is unclear to me what you mean with "the owner" (i don't believe mqtt messages have owner attributes)

**Review by evils — COMMENTED** [review]
thanks

File: README.md (hunk 8 above)
Comment (friedkiwi) [reply README.md:33]: owner here, as in, the person who called. Then this could be used as a source for eg flappers.

Comment (evils) [reply README.md:33]: ah, maybe something like "upon a successful call to open the gate, the `owner` field from the whitelist will be published here"

**Pull request merged**